File: lib/engines/ai-summarizer.ts
import { OpenAI } from 'openai';

const openai = new OpenAI({ apiKey: process.env.OPENAI_KEY });

export async function summarizeFact(claim: string, sources: any[]) {
  const prompt = `Claim: ${claim}
Sources: ${sources.map(s=>s.url).join(', ')}
Summarize verdict and confidence.`;
  const resp = await openai.chat.completions.create({
    model: "gpt-4",
    messages: [{ role: "system", content: prompt }]
  });
  // Parse resp to extract summary, verdict, confidence
  return {
    status: "True", // or "False"/"Uncertain"
    confidence: 95,
    summary: resp.choices[0].message.content
  };
}

File: lib/engines/ai-summarizer.py
from openai import OpenAI

openai = OpenAI(api_key="YOUR_OPENAI_KEY")

def summarize_fact(claim, sources):
    """
    Summarize the fact-check result in 30-60 words, include contradictions and confidence.
    """
    prompt = (
        f"Claim: {claim}
"
        f"Sources: {', '.join(s['url'] for s in sources)}
"
        "Summarize the truth, contradictions, and confidence in 30-60 words, for podcast listeners."
    )
    response = openai.chat.completions.create(
        model="gpt-4",
        messages=[{"role": "system", "content": prompt}]
    )
    return response.choices[0].message.content.strip()